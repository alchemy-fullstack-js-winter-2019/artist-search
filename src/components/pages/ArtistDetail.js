import React, { PureComponent } from 'react';
import { getArtistDetail } from '../services/artistApi';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default class ArtistDetail extends PureComponent {
  state = {
    works: []
  }
  static propTypes = {
    match: PropTypes.object
  }

  // fetchArtists = async() => {
  //   const works = await getArtistDetail(this.props.match.params.id);
  //   this.setState({ works });
  // };

  componentDidMount() {
    getArtistDetail(this.props.match.params.id)
      .then(res => {
        this.setState({ works: res.results });
      });
    // this.fetchArtists();
  }


  render() {
    const works = this.state.works.map(work => {
      return (
        <Link key={work.id} to={`/artistDetail/${work.id}`}><li>{work.title}</li></Link>
      );
    });
    return (
      <ul>
        { works }
      </ul>
    );
  }
}
