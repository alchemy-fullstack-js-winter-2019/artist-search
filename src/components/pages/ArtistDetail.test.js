import React from 'react';
import { shallow } from 'enzyme';
import ArtistDetail from './ArtistDetail';

jest.mock('../services/artistApi');

describe('Artists', () => {
  it('renders a list of artist works', () => {
    const wrapper = shallow(<ArtistDetail 
      match={{ params: { id: '9282c8b4-ca0b-4c6b-b7e3-4f7762dfc4d6' } }}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
