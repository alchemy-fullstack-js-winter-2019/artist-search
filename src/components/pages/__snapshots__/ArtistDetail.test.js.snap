// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists renders a list of artist works 1`] = `<ul />`;
